### PR TITLE
[processor/deltatocumulative]: timer-based expiry

### LINKED ISSUE #31615: [processor/deltatocumulative] internal/streams.TestExpiry fails on Windows
### Component(s)

processor/deltatocumulative

### What happened?

TestExpiry fails on Windows. See https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8165426471/job/22322842022

```
=== Failed
=== FAIL: internal/streams TestExpiry (0.78s)
    expiry_test.go:81: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/streams/expiry_test.go:81
        	Error:      	Not equal: 
        	            	expected: 1m0s
        	            	actual  : 1m1s
        	Test:       	TestExpiry

=== FAIL: internal/streams TestExpiry (re-run 1) (0.82s)
    expiry_test.go:81: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/internal/streams/expiry_test.go:81
        	Error:      	Not equal: 
        	            	expected: 1m0s
        	            	actual  : 1m1s
        	Test:       	TestExpiry

DONE 2 runs, 14 tests, 2 failures in 15.740s
make[2]: *** [../../Makefile.Common:126: test] Error 1
make[1]: *** [Makefile:165: processor/deltatocumulativeprocessor] Error 2
make[1]: *** Waiting for unfinished jobs....
```

### PR DESCRIPTION
**Pull request by tombrk**
**Description:** Moves from complex preemptive expiry to a plain 1 minute timer 

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31615#issuecomment-1980758429

Resolves https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31615

### CODE REVIEW

File: processor/deltatocumulativeprocessor/processor.go [hunk 1]
```diff
@@ -29,8 +31,8 @@ type Processor struct {
 	ctx    context.Context
 	cancel context.CancelFunc
 
-	aggr streams.Aggregator[data.Number]
-	exp  *streams.Expiry[data.Number]
+	aggr  streams.Aggregator[data.Number]
+	stale *staleness.Staleness[data.Number]
```
Comment (RichieSams) [inline processor/deltatocumulativeprocessor/processor.go:35]: minor: You could name this say, `nums` in preparation for having multiple. (IE for histograms, etc)

**Comment by RichieSams** [comment]
Added a minor naming comment. Otherwise LGTM :)

File: processor/deltatocumulativeprocessor/processor.go (hunk 1 above)
Comment (sh0rez) [reply processor/deltatocumulativeprocessor/processor.go:35]: let's do that when its needed. I'd like to experiment with the type system more if we can't just drop the datapoint narrowing on the aggregator altogether

**Comment by crobert-1** [comment]
Minor nit: @sh0rez can you update the issue description to say "Resolves #31615"? This will allow our GitHub automation to automatically close the issue (reducing the chance of us forgetting) 👍

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
Thank you, @sh0rez!

**Pull request merged**